packages/common/pitop/common/ptdm.py

```diff
@@ -457,35 +459,37 @@ def __thread_method(self):
         poller.register(self._zmq_socket, zmq.POLLIN)
         while self.__continue:
             events = poller.poll(_TIMEOUT_MS)
-
             for _ in range(len(events)):
                 message_string = self._zmq_socket.recv_string()
                 message = Message.from_string(message_string)
 
-                id = message.message_id()
-                if id in self._callback_funcs:
-                    self.invoke_callback_func_if_exists(
-                        self._callback_funcs[id], message.parameters
-                    )
-
-    def invoke_callback_func_if_exists(self, func, params=list()):
-        if not callable(func):
-            return
-
-        func_arg_no = len(signature(func).parameters)
-        if func_arg_no > 1:
-            logger.error(
-                "Invalid callback function - it should receive at most one argument."
-            )
-            return ""
+                callback = self._callback_funcs.get(message.message_id())
+                if callback:
+                    self.invoke_callback_func_if_exists(callback, message.parameters)
 
+    def invoke_callback_func_if_exists(self, callback, params=list()):
```

Comment:
i'm not sure but this looks like it would have the same issue as using `[]` as a default, that the same list is actually reused each time. probably not an issue since you don't mutate it, but I still think it's better practice to use None as the default and then have like `params = list() if params is None else params`

https://nikos7am.com/posts/mutable-default-arguments/

Reply:
also, not new things but:
- the naming of this function is confusing me - it doesn't check wether the callback exists it just calls it?
- below is `if params == list()` just a weird way to check if the list is empty? usually people check the length?